src/serial/private/utils/list_serialports_windows.nim

```diff
@@ -44,10 +44,30 @@ proc SetupDiDestroyDeviceInfoList(DeviceInfoSet: HDEVINFO): bool
 
 iterator listSerialPorts*(): string =
   ## Iterates through a list of serial port names currently available on the system.
-  ##
-  ## This is based upon the `CEnumerateSerial::QueryUsingSetupAPI` method from `CEnumerateSerial`: http://www.naughter.com/enumser.html
 
-  # First create a "device info set" for the device interface GUID
+  
+  ## First we enumerate the USB serial ports from the appropritate registry node, 
+  ## this is because some of them are reported not to show up using the second method below.
+  ## see https://github.com/euantorano/serial.nim/issues/26
+  var usbSerialPortList HashSet[string]
```

Comment:
Seems to be missing a `:` here:

```suggestion
  var usbSerialPortList: HashSet[string]
```

Reply:
Hi, thanks for leaving it open, I WAS already working on it :grin: , and you are right, it is MUCH cleaner. Please see the updates to this pull request.  Please especially take a look at the updates to the windows_registry module, as that is where most of the work went.

Reply:
You're right, this is a LOT nicer and easier to follow. Thanks very much!